packages/waku/src/lib/utils/vite-loader.ts

```diff
@@ -1,11 +1,13 @@
 import { createServer as createViteServer } from 'vite';
 import { fileURLToFilePath } from '../utils/path.js';
+import { typegenPlugin } from '../plugins/vite-plugin-typegen.js';
 
 export const loadServerFile = async (fileURL: string) => {
   const vite = await createViteServer({
     ssr: {
       external: ['waku'],
     },
+    plugins: [typegenPlugin()],
```

Comment:
Can we remove this?

Reply:
This is needed to run the generation step with `waku dev` - Ideally we'd include this here and I figure out how to run the generation in watch mode if you're ok with that.

Reply:
I think it should be in dev-server-impl.ts.

This file is an unfortunate hack, and we would like to remove it in the future.

Reply:
oh! moving this to there fixes the hot reload issues :) thanks for the tip. I'll polish and have something more reviewable in the next couple of days